frontend/src/locale/context.tsx

```diff
@@ -22,8 +24,11 @@ export const LocaleProvider: React.FC<{ lang: string }> = ({
   children,
 }) => {
   React.useEffect(() => {
-    if (lang !== localStorage.getItem("locale")) {
-      localStorage.setItem("locale", lang);
+    const { pyconLocale } = cookies({
+      req: { headers: { cookie: document.cookie } },
+    });
+    if (lang !== pyconLocale) {
+      document.cookie = `pyconLocale=${lang}; path=/`;
```

Comment:
why are you using cookies for the language?

Reply:
we can access them on the server and redirect server side

Reply:
otherwise we would have to load the JS, flash the UI for a second and then redirect to the correct path

Reply:
ok, just saw it below :) we don't use the user language by default anymore?

Reply:
I wanted to add it back in another PR so keep this one simple, I can do it in this one too

Reply:
It’s ok to do in another PR :)